_release/bat/base_bat/base_bat_test.go

```diff
@@ -311,6 +327,169 @@ func TestDeleteAllInstances(t *testing.T) {
+		img.Visibility != "public" || img.Protected {
+		t.Errorf("Meta data of added image is incorrect")
+	}
+	//get data to create workload yaml file
+	defaults := bat.DefaultResources{
+		VCPUs: 2,
+		MemMB: 128,
+	}
+
+	source := bat.Source{
+		Type: "image",
+		ID:   img.ID,
+	}
+	disk := bat.Disk{
+		Bootable:  true,
+		Source:    &source,
+		Ephemeral: true,
+	}
+
+	WLopts := bat.WorkloadOptions{
+		Description:     "BAT VM Test",
+		VMType:          "qemu",
+		FWType:          "legacy",
+		Defaults:        defaults,
+		CloudConfigFile: "CirrosInit.yaml",
+		Disks:           []bat.Disk{disk},
+	}
+	y, err := yaml.Marshal(WLopts)
+	if err != nil {
+		t.Fatal("fail to read yaml file $v", err)
+	}
+	ioutil.WriteFile("CirrosWorkload.yaml", y, 0644)
+	ioutil.WriteFile("CirrosInit.yaml", []byte(vmCloudInit), 0644)
+
+	//create the workload
+	args := []string{"workload", "create", "-yaml", "CirrosWorkload.yaml"}
+	_, err = bat.RunCIAOCLIAsAdmin(ctx, "", args)
+	if err != nil {
+		t.Fatal("Fail to create workload $v", err)
+
+	}
+	//Get ID of workload just created
+	CurWL, err := bat.GetWorkloadByName(ctx, "", WLopts.Description)
+	if err != nil {
+		t.Fatalf("Can't find data for workload %v", err)
+	}
+	//Launch instance of that workload
+	instance, err := bat.LaunchInstances(ctx, "", CurWL.ID, 1)
+	if err != nil {
+		t.Fatal("Fail to launch instance: $v", err)
+	}
+	_, err = bat.WaitForInstancesLaunch(ctx, "", instance, false)
+	if err != nil {
+		t.Errorf("Instance %s did not launch: %v", instance[0], err)
+	}
+	//Access the instances with the user and pasword specified in the could init file
+	config := &ssh.ClientConfig{
+		User: "ciaouser",
+		Auth: []ssh.AuthMethod{
+			ssh.Password("FrakbS}k0F5Xm0B"),
+		},
+	}
+	Template := `{{$x := filterContains . "Status" "active"}}{{range $x}}{{.SSHIP}}:{{.SSHPort}}{{end}}`
+	argsInstance := []string{"instance", "list", "-f", Template}
+	InstanceIP, err := bat.RunCIAOCLIAsAdmin(ctx, "", argsInstance)
+	if err != nil {
+		t.Fatal("Fail to create get Instance's data $v", err)
+	}
+
+	client, err := ssh.Dial("tcp", string(InstanceIP[:]), config)
+	if err != nil {
+		t.Fatal("Failed to dial: " + err.Error())
+	}
+	session, err := client.NewSession()
+	if err != nil {
+		t.Fatal("Failed to create session: " + err.Error())
+	}
+	defer session.Close()
+
+	var b bytes.Buffer
+	session.Stdout = &b
+	if err := session.Run("/usr/bin/whoami"); err != nil {
+		t.Fatal("Failed to run: " + err.Error())
+	}
+
+	//Delete all instances
+	_, err = bat.DeleteInstances(ctx, "", instance)
+	if err != nil {
+		t.Errorf("Failed to delete instance: %v", err)
+	}
+
+	//Pending delete workload until function implemented
+}
+
+//Cleanup process tu run after TestLaunchCustomInstance
+//
+//Delete the cirros***.img file downloaded, the yaml files for the image
+//created, the image and the instance
+//
+//all data created for the test is removed
+func deleteCustomData(ctx context.Context, ID string) {
```

Comment:
Actually, only exported functions beginning with Test need to be commented in this way.  These comments don't hurt though.